src/rpc/rpc-handlers.ts

```diff
@@ -276,10 +257,6 @@ const handleRPC: HandlerFunc = async data => {
             const res = await permissionHandler({id, method, params, origin, network});
             return res;
         }
-        if (method.startsWith('keplr_')) {
-            const specialHandler = specialHandlers['keplr'];
-            return specialHandler({id, method, params, network, target, contract});
-        }
```

Comment:
Instead of treating all `keplr_...` methods as special, we could explicitly list the _special_ ones in the `specialHandlers` as I've done above, and let the rest be handled by the `send` method I mentioned in a previous comment.

By _special_ in this context we mean methods that require user confirmation, and that means methods that write to the blockchain, hence requiring users to spend funds.

With this change, the flow is the same for the three providers:
* a request comes in
* we check if the method has a _specialHandler_
* if it does, we call the _specialHandler_
* if it doesn't, we call the `send` method for the given provider and immediately send the tx to the blockchain

Reply:
this handler handles all the cases, it's not using rpc so old structure has no sense